cmd/git/main_test.go

```diff
@@ -429,4 +430,137 @@ var _ = Describe("Git Resource", func() {
 			})
 		})
 	})
+
+	Context("failure diagnostics", func() {
+		const exampleSSHGithubRepo = "git@github.com:shipwright-io/sample-go.git"
+		const nonExistingSSHGithubRepo = "git@github.com:shipwright-io/sample-go-nonexistent.git"
+		const exampleHTTPGithubNonExistent = "https://github.com/shipwright-io/sample-go-nonexistent.git"
+		const githubHTTPRepo = "https://github.com/shipwright-io/sample-go.git"
+
+		const exampleSSHGitlabRepo = "git@gitlab.com:gitlab-org/gitlab-runner.git"
+		const exampleHTTPGitlabNonExistent = "https://gitlab.com/gitlab-org/gitlab-runner-nonexistent.git"
+		const gitlabHTTPRepo = "https://gitlab.com/gitlab-org/gitlab-runner.git"
+
+		var sshPrivateKey string
+
+		BeforeEach(func() {
+			if sshPrivateKey = os.Getenv("TEST_GIT_PRIVATE_SSH_KEY"); sshPrivateKey == "" {
+				Skip("Skipping private repository tests since TEST_GIT_PRIVATE_SSH_KEY environment variable is not set")
+			}
+		})
```

Comment:
May you move this skip logic to the block that really requires the sshPrivateKey. Should be only `should detect non-existing repo given ssh authentication`.

Reply:
good point thanks